src/p_mobj.c

```diff
@@ -799,7 +833,7 @@ void P_MobjThinker (mobj_t* mobj)
 // P_SpawnMobj
 //
 
-mobj_t *P_SpawnMobj(fixed_t x, fixed_t y, fixed_t z, mobjtype_t type)
+mobj_t *P_SpawnMobjSafe(fixed_t x, fixed_t y, fixed_t z, mobjtype_t type, boolean safe)
```

Comment:
When I see "safe" in a function name I usually assume it has extra handling to avoid crashing or something similar. It's a bit ambiguous for this case but it's not so hard to figure out looking at the code, so maybe this is fine. Maybe something like `P_SpawnMobj2(..., boolean cosmetic)`? Idk about the function name but cosmetic as a variable may be more clear.

Reply:
Hmm, I am not sure about that. This boolean fixes demo compatibility and is not responsible for any cosmetic changes. Maybe you could come up with some other more suitable and descriptive variable name?

Reply:
This variable makes the result of the method purely cosmetic (not affecting game state). As far as I can tell, that is the key distinction, and what you mean by "safe".